sbt-version-policy/src/main/scala/sbtversionpolicy/DependencyCheckReport.scala

```diff
@@ -61,44 +62,120 @@ object DependencyCheckReport {
+    currentModules: Map[(String, String), String],
+    previousModules: Map[(String, String), String],
+    reconciliations: Seq[(ModuleMatchers, VersionCompatibility)],
+    defaultReconciliation: VersionCompatibility
   ): DependencyCheckReport = {
 
-    val backward = moduleStatuses(currentModules, previousModules, reconciliations, defaultReconciliation)
-    val forward = moduleStatuses(currentModules, previousModules, reconciliations, defaultReconciliation)
+    // FIXME These two lines compute the same result. What is the reason for having two directions?
+    val backward = moduleStatuses(compatibilityIntention, currentModules, previousModules, reconciliations, defaultReconciliation)
+    val forward = moduleStatuses(compatibilityIntention, currentModules, previousModules, reconciliations, defaultReconciliation)
 
     DependencyCheckReport(backward, forward)
   }
 
+  @deprecated("Internal", "1.1.0")
   def moduleStatuses(
     currentModules: Map[(String, String), String],
     previousModules: Map[(String, String), String],
     reconciliations: Seq[(ModuleMatchers, VersionCompatibility)],
     defaultReconciliation: VersionCompatibility
   ): Map[(String, String), ModuleStatus] =
-    for ((orgName @ (org, name), ver) <- previousModules) yield {
+    moduleStatuses(
+      Compatibility.BinaryCompatible,
+      currentModules,
+      previousModules,
+      reconciliations,
+      defaultReconciliation
+    )
+
+  private def moduleStatuses(
+    compatibilityIntention: Compatibility,
+    currentModules: Map[(String, String), String],
+    previousModules: Map[(String, String), String],
+    reconciliations: Seq[(ModuleMatchers, VersionCompatibility)],
+    defaultReconciliation: VersionCompatibility
+  ): Map[(String, String), ModuleStatus] =
+    for ((orgAndName @ (org, name), previousVersion) <- previousModules) yield {
 
-      val status = currentModules.get(orgName) match {
-        case None => Missing(ver)
-        case Some(`ver`) => SameVersion(ver)
+      val status = currentModules.get(orgAndName) match {
+        case None => Missing(previousVersion)
+        case Some(`previousVersion`) => SameVersion(previousVersion)
         case Some(currentVersion) =>
           val reconciliation = reconciliations
             .collectFirst {
               case (matcher, rec) if matcher.matches(org, name) =>
                 rec
             }
             .getOrElse(defaultReconciliation)
-          if (reconciliation.isCompatible(ver, currentVersion))
-            CompatibleVersion(currentVersion, ver, reconciliation)
+          val isCompatible =
+            if (compatibilityIntention == Compatibility.BinaryAndSourceCompatible) {
+              isBinaryCompatible(currentVersion, previousVersion, reconciliation) &&
+                isSourceCompatible(currentVersion, previousVersion, reconciliation)
+            } else {
+              isBinaryCompatible(currentVersion, previousVersion, reconciliation)
+            }
```

Comment:
According to the intended compatibility, we check the source compatibility or not.